backend/src/services/auth/auth.service.ts

```diff
@@ -31,4 +34,102 @@ const signupLocal = async (user: UserCreateInput): Promise<AuthResponseDto> => {
   };
 };
 
-export { signupLocal };
+const requestPasswordReset = async (email: string): Promise<string> => {
+  const user = await prisma.user.findUnique({
+    where: { email },
+  });
+  if (!user) throw new Error('User with such email does not exist');
+
+  const resetToken = jwt.sign(
+    {
+      id: user.id,
+    },
+    ENV.JWT.SECRET!,
+    { expiresIn: '1h' },
+  );
+
+  await prisma.user_Security.update({
+    where: { user_id: user.id },
+    data: { password_change_token: resetToken },
+  });
+
+  const link = `localhost:${ENV.APP.SERVER_PORT}${ENV.API.V1_PREFIX}/auth/local/reset-password?token=${resetToken}`;
+
+  sendEmail(
+    user.email,
+    'Password Reset Request',
+    {
+      name: user.name,
+      link: link,
+    },
+    './template/requestResetPassword.handlebars',
+  );
+  return link;
+};
+
+const resetPasswordCheckToken = async (token: string): Promise<string> => {
+  const payload = jwt.verify(token, ENV.JWT.SECRET!) as jwt.JwtPayload;
+  const userSecurity = await prisma.user_Security.findUnique({
+    where: { user_id: payload.id },
+  });
+
+  if (!userSecurity) {
+    throw new Error('User does not exist');
+  }
+  if (!userSecurity.password_change_token) {
+    throw new Error('Password reset token does not exist');
+  }
+  if (token !== userSecurity.password_change_token) {
+    await prisma.user_Security.update({
+      where: { user_id: payload.id },
+      data: { password_change_token: null },
+    });
+    throw new Error('Invalid password reset token');
+  }
+
+  await prisma.user_Security.update({
+    where: { user_id: payload.id },
+    data: { password_change_token: null },
+  });
+
+  return payload.id;
```

Comment:
The duplication of the token deletion can be avoided like this: 

```suggestion
  // Resets password change token either if activation was successful or failed
  await prisma.user_Security.update({
    where: { user_id: payload.id },
    data: { password_change_token: null },
  });
  
  if (token !== userSecurity.password_change_token) {
    throw new Error('Invalid password reset token');
  }

  return payload.id;
```